### ISSUE TITLE
Support for regex tokenizer

### ISSUE THREAD

**Issue by guilload** [quickwit-oss/quickwit#2652]
A regex tokenizer is now available in tantivy.

```yaml
field_mappings:
  - name: message
    type: text
    tokenizer: regex # or "pattern" à la ES?
    pattern: \W+
```

**Comment by fulmicoton** [comment]
We do not support groups yet. External contributors are working on this.

**Comment by evanxg852000** [comment]
I have started this but not sure what you mean by:
> We do not support groups yet ...
are you talking about grouping as in the regex syntax support for grouping?

**Comment by fmassot** [comment]
> are you talking about grouping as in the regex syntax support for grouping?
yes exactly

**Comment by fulmicoton** [comment]
Yes. Group as in regex syntax to support grouping. 
The idea is that you select a group as the token that should be extracted.

This is something supported by elastic.
https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-pattern-tokenizer.html

Our current RegexTokenizer does not support it.

**Comment by fmassot** [comment]
closed by #3056
